pkg/providers/v2/cloud.go

```diff
@@ -61,6 +78,30 @@ type EC2Metadata interface {
 	GetMetadata(path string) (string, error)
 }
 
+func readAWSCloudConfig(config io.Reader) (*v1alpha1.AWSCloudConfig, error) {
```

Comment:
Can we add a unit test for this? Here's a similar example: https://github.com/kubernetes/kubernetes/blob/5344afd4fbfc3cb2ed6785d89b6f760886f100b5/pkg/credentialprovider/plugin/config_test.go#L30-L255 

I think you would have to initiate a buffer (bytes.NewBuffer) with the file contents, as opposed to passing the file name but the rest of the test logic should be similar.

Reply:
gotcha, will update!